applications/GeoMechanicsApplication/geo_mechanics_application_constants.h

```diff
@@ -0,0 +1,87 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Vahid Galavi
+//
+
+#if !defined(KRATOS_GEO_MECHANICS_APPLICATION_CONSTANTS_H_INCLUDED )
+#define  KRATOS_GEO_MECHANICS_APPLICATION_CONSTANTS_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+
+
+namespace Kratos
+{
+    /* Contants used in GeoMechanicsApplication */
+    // The size type definition
+    typedef std::size_t         SizeType;
+
+
+    // Static definition of the dimension
+    constexpr SizeType N_DIM_3D = 3;
+    constexpr SizeType N_DIM_2D = 2;
+    constexpr SizeType N_DIM_1D = 1;
+
+    // Limits
+    constexpr double TINY  = 1.0e-60;
+    constexpr double LARGE = 1.0e10;
+
+    // factor for pore pressure calculations: 1: mechanical sign convention, -1: soil mechanics sign convention
+    constexpr double PORE_PRESSURE_SIGN_FACTOR = 1.0;
+
+    // Static definition of the VoigtSize
+    constexpr SizeType VOIGT_SIZE_3D = 6;
+    constexpr SizeType VOIGT_SIZE_2D_PLANE_STRESS = 3;
+    constexpr SizeType VOIGT_SIZE_2D_PLANE_STRAIN = 4; // This needs to be changed
+    constexpr SizeType VOIGT_SIZE_2D_AXISYMMETRIC = 4;
+    constexpr SizeType VOIGT_SIZE_2D_INTERFACE    = 2;
+    constexpr SizeType VOIGT_SIZE_3D_INTERFACE    = 3;
+
+    // stress/strain vector indices (3D):
+    enum indexStress3D: int{ INDEX_3D_XX,
+                             INDEX_3D_YY,
+                             INDEX_3D_ZZ,
+                             INDEX_3D_XY,
+                             INDEX_3D_YZ,
+                             INDEX_3D_XZ };
+
+    // stress/strain vector indices (2D plane strain):
+    enum indexStress2DPlaneStrain: int{ INDEX_2D_PLANE_STRAIN_XX,
```

Comment:
```suggestion
    enum class indexStress2DPlaneStrain{ INDEX_2D_PLANE_STRAIN_XX,
```
more C++11

Reply:
I tried but this creates a lot of errors. I leave it as it is for the time being.